tests/test_the_test/test_minimal_number_of_scenarios.py

```diff
@@ -0,0 +1,83 @@
+import re
+import warnings
+from utils import scenarios
+from utils._context._scenarios import get_all_scenarios
+from utils._context._scenarios.endtoend import EndToEndScenario
+
+
+@scenarios.test_the_test
+def test_minimal_number_of_scenarios():
+    """
+    Verify that EndToEnd scenarios don't have identical environment variables.
+    If two scenarios have the same weblog_env and agent_env, they could be merged.
+    """
+    all_scenarios = get_all_scenarios()
+    
+    # Filter only scenarios that are exactly EndToEndScenario class (not subclasses)
+    endtoend_scenarios = [
+        scenario for scenario in all_scenarios 
+        if type(scenario) == EndToEndScenario
+    ]
+    
+    def normalize_env(env_dict):
+        """Normalize environment dict by removing dynamic/system variables and handling None values."""
+        if not env_dict:
+            return {}
+        
+        # Exclude variables that are dynamic or system-specific
+        excluded_vars = {
+            'SYSTEMTESTS_SCENARIO',  # This is always different per scenario
+            'DD_LOGS_INJECTION',     # May be added dynamically
+        }
+        
+        normalized = {}
+        for key, value in env_dict.items():
+            if key not in excluded_vars:
+                # Convert None values to empty string for consistent comparison
+                normalized[key] = str(value) if value is not None else ""
```

Comment:
Setting `None` in env removes the value if it has been defined, so I think we cannot convert None to `""`

Reply:
FYI, Code to play with the SDK, to see how it behaves : 

```python
import docker

client = docker.from_env()
output = client.containers.run(
    "alpine",
    "env",
    environment={"HOSTNAME": None, "FOO": ""},  # removes hostname, declare foo as enpty string
    remove=True
)
print(output.decode())
```